vignettes/data-transform-as-shiny-module.Rmd

```diff
@@ -160,6 +160,70 @@ if (interactive()) {
 When a custom transformation is used, the UI for the transformation is placed below the filter panel.
 However, there is a way to customize the placement of the UI inside the module content.
 
-This can only be done when the module allocates a dedicated space for the transform UI.
+In order to place the transformation UI inside the module there are few things one has to do:
+1. Create a custom module wrapper function.
+2. Call the desired module in the module wrapper function and store it in a variable so it can be modified.
+3. Modify the UI of the module with the transform UI at the desired location by calling the `ui_transform_data`. Note that in order for the transform to work you need to change the name space of the id by passing `NS(gsub("-module$", "", id), "data_transform")`.
```

Comment:
@averissimo The namespace change is the part that is not great. Is there another way to do this?
@m7pr suggested to have some kind of a wrapper to achieve this. Either for the name space or the whole module customization itself.

Reply:
I think there needs to be a trick somewhere as we're injecting the `ui_transform` under different namespace locations, so we have 2 options:

1. As is with no namespace trick when injecting the `ui_transform_data` in most cases and trick on manual
2. The other way round

From a design point of view, it seems that the trick should be applied on the exception/rare case. But this makes it module-developer visible.

Feel more than free to propose a change :-D ... I'm not keen on it as it's a lesser of evils kind of implementation.

I've tried getting to the `SessionProxy` parent, but the public API seems to not allow for it, `override` of `SessionProxy` is lost as scope is created (only accessing the original session)

Reply:
Would this help a bit, if we expose/create such a function in this example?
```
transform_id <- function(id) {
   NS(gsub("-module$", "", id), "data_transform")
}
```

Reply:
`transform_id` does simplify getting the namespace with the added advantage that we can change the namespaces in teal and still not break apps using this feature because they use our wrapper.
Are we okay to export a function that caters to a small audience? Advanced users should be okay with this kind of complexity, moreover, we will have teal transform wrappers and this could be a topic to discuss during that time.
And this feature could be like a hidden feature that we don't advertise.